homeassistant/components/modem_callerid/sensor.py

```diff
@@ -11,111 +10,112 @@
     EVENT_HOMEASSISTANT_STOP,
     STATE_IDLE,
 )
-import homeassistant.helpers.config_validation as cv
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv, entity_platform
+
+from .const import (
+    DATA_KEY_API,
+    DEFAULT_DEVICE,
+    DEFAULT_NAME,
+    DOMAIN,
+    ICON,
+    SERVICE_REJECT_CALL,
+    STATE_CALLERID,
+    STATE_RING,
+)
 
-_LOGGER = logging.getLogger(__name__)
-DEFAULT_NAME = "Modem CallerID"
-ICON = "mdi:phone-classic"
-DEFAULT_DEVICE = "/dev/ttyACM0"
-
-STATE_RING = "ring"
-STATE_CALLERID = "callerid"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
-    }
+PLATFORM_SCHEMA = cv.deprecated(
+    vol.All(
+        PLATFORM_SCHEMA.extend(
+            {
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
+            }
+        )
+    )
 )
 
+_LOGGER = logging.getLogger(__name__)
 
```

Comment:
You could change `setup_platform` to `async_setup_platform` and do the import from there, that way you can eliminate `async_setup` in `__init__.py` and you can be sure the config has been validated.

Reply:
Where is `setup_platform` in this PR? 😕

Reply:
https://github.com/home-assistant/core/pull/46677/files#diff-f6858dc32f5966fb33239da9708297f60bc75ad647cd6bba5d41d6b13f910f0cL32

Reply:
Should i use async_create_task or async_add_entities?

Reply:
example here https://github.com/home-assistant/core/blob/2021.1.5/homeassistant/components/harmony/remote.py#L104